.github/workflows/publish.yml

```diff
@@ -0,0 +1,202 @@
+---
+name: Publish to pypi
+on:
+  push:
+  #On versioned releases
+    tags:
+      - '*.*.*'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      force:
+        type: choice
+        description: Retry Publish Version
+        options:
+          - No
+          - Yes
+      environment:
+        description: 'Deployment environment'
+        required: true
+        default: 'pypi'
+        type: choice
+        options:
+          - pypi
+          - testpypi
+      dryRun:
+        description: 'Dry Run deployment (set to false to deploy)'
+        required: true
+        type: boolean
+        default: true
+
+
+
+jobs:
+  lint:
+    if: github.repository == 'ckan/ckanext-validation'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install requirements
+        run: pip install flake8 pycodestyle
+      - name: Check syntax
+        run: flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics --extend-exclude ckan
+
+  test:
+    needs: lint
+    strategy:
+      matrix:
+        include: #ckan-image see https://github.com/ckan/ckan-docker-base, ckan-version controls other image tags
+          - ckan-version: "2.11"
+            ckan-image: "2.11-py3.10"
```

Comment:
Looks like our Jinja version can't handle Python 3.10, which moved `Mapping` from `collections` to `collections.abc`